### PR TITLE
[Feature] Soft wing loss (TIP'2021)

### PR DESCRIPTION
**Pull request by jin-s13**
## Motivation

Support SoftWingLoss. 
"Structure-Coherent Deep Feature Learning for Robust Face Alignment" TIP'2021

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmpose/models/losses/regression_loss.py [hunk 1]
```diff
@@ -117,6 +118,84 @@ def forward(self, output, target, target_weight=None):
         return loss * self.loss_weight
 
 
+@LOSSES.register_module()
+class SoftWingLoss(nn.Module):
+    """Soft Wing Loss 'Structure-Coherent Deep Feature Learning for Robust Face
```
Comment (ly015) [inline mmpose/models/losses/regression_loss.py:123]: Please follow the standard reference format:

https://github.com/open-mmlab/mmpose/blob/228747c298c0801dea093371abeca539d41a7584/mmpose/models/backbones/hrnet.py#L218-L219

**Review by ly015 — APPROVED** [review]
